### PR TITLE
[GC-5057] Clean up remaining plugin checker errors

### PR DESCRIPTION
**Pull request by alexoakland**
### Clean up remaining plugin checker errors
Link to ticket: https://bynder.atlassian.net/browse/GC-5057

### What has been done
- Cleaned up various remaining errors found by the plugin checker tool 

The only errors that remain are translation related errors which require a comment to help any translators create a translation, and then various errors relating to hidden files and application files being included in the plugin which is likely just the result of mounting the code in the container.

**THIS HAS VARIOUS TICKETS MERGED INTO IT SO WILL BE MUCH EASIER TO REVIEW WHEN OTHERS ARE MERGED IN**

### CODE REVIEW

File: includes/classes/admin/mapping/field-types/acf.php [hunk 1]
```diff
@@ -52,8 +53,13 @@ public function underscore_template( View $view ) {
 			}, $groupIds ) );
 
 			// Prepare and execute query to get all fields for all groups
-			$fields_query   = "SELECT * FROM {$wpdb->posts} WHERE post_type = 'acf-field' AND post_content LIKE '%repeater%' AND post_parent IN ($groupIdPlaceholders)";
-			$fields_results = $wpdb->get_results( $wpdb->prepare( $fields_query, $groupIds ) );
+			$fields_results = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT * FROM {$wpdb->posts} WHERE post_type = 'acf-field' AND post_content LIKE %s AND post_parent IN ($groupIdPlaceholders)",
+					'%' . $wpdb->esc_like('repeater') . '%',
```
Comment (CWDN) [inline includes/classes/admin/mapping/field-types/acf.php:59]: I don't get this change, wasn't `repeater` just a hardcoded string anyway, way does it need to be escaped?

File: includes/classes/debug.php [hunk 2]
```diff
@@ -296,8 +296,12 @@ public function do_debug_options_actions( $settings ) {
 	public function handle_stuck_statuses( $settings, $back_button ) {
 		global $wpdb;
 
-		$sql     = "SELECT `option_name` FROM `$wpdb->options` WHERE `option_name` LIKE ('gc_pull_item_%') OR `option_name` LIKE ('gc_push_item_%');";
-		$options = $wpdb->get_results( $sql );
+		$options = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT `option_name` FROM %s WHERE `option_name` LIKE ('gc_pull_item_%%') OR `option_name` LIKE ('gc_push_item_%%');",
+				$wpdb->options
```
Comment (CWDN) [inline includes/classes/debug.php]: Does this definitely work as I had problems with doing prepared statements and the table name, it never seemed to be able to find the table?

File: includes/classes/admin/mapping-wizard.php [hunk 3]
```diff
@@ -151,11 +151,15 @@ public function add_help_tabs() {
...
-				             ) . '<p>' . $this->refresh_connection_link() . '</p>',
+				'content' => sprintf(
+					__('<p>Thank you for using the Content Workflow WordPress plugin!</p><p>To make the plugin more speedy, we cache the requests to GatherContent for 1 day, but if you find that you need to update the data from Content Workflow, just hit the "Refresh" button.</p>%s', 'content-workflow-by-bynder'),
+					'<p>' . $this->refresh_connection_link() . '</p>'
+				),
+//				'content' => __(
+//					             '<p>Thank you for using the Content Workflow WordPress plugin!</p>' .
+//					             '<p>To make the plugin more speedy, we cache the requests to GatherContent for 1 day, but if you find that you need to update the data from Content Workflow, just hit the "Refresh" button.</p>',
+//					             'content-workflow-by-bynder'
+//				             ) . '<p>' . $this->refresh_connection_link() . '</p>',
```
Comment (CWDN) [inline includes/classes/admin/mapping-wizard.php]: Do we need the commented out version now?

File: includes/classes/sync/pull.php [hunk 4]
```diff
@@ -812,7 +811,12 @@ protected function set_acf_field_value( $post_data ) {
 												$file = $upload_dir['basedir'] . '/' . $filename;
 											}
 
-											file_put_contents( $file, $image_data );
+											require_once( ABSPATH . 'wp-admin/includes/file.php' );
+											WP_Filesystem();
+											global $wp_filesystem;
+											$wp_filesystem->put_contents( $file, $image_data );
+
+											// file_put_contents( $file, $image_data );
```
Comment (CWDN) [inline includes/classes/sync/pull.php]:
```suggestion
```

File: includes/classes/admin/mapping/field-types/acf.php (hunk 1 above)
Comment (alexoakland) [reply includes/classes/admin/mapping/field-types/acf.php:59]: It's a suggestion from the plugin checker.

**ERROR**: WordPress.DB.PreparedSQLPlaceholders.LikeWildcardsInQuery
**Message**: SQL wildcards for a LIKE query should be passed in through a replacement parameter. Found: LIKE '%repeater%'.

File: includes/classes/debug.php (hunk 2 above)
Comment (alexoakland) [reply includes/classes/debug.php]: Didn't work! Whacked an ignore in!

File: includes/classes/sync/push.php [hunk 5]
```diff
@@ -604,7 +604,7 @@ protected function set_database_field_value( $tableColumnString ) {
 
 		global $wpdb;
 
-		$results = $wpdb->get_results( $wpdb->prepare( "SELECT %i as value from %i where post_id=%d;", $column, $table, $this->post->ID ) );
+		$results = $wpdb->get_results( $wpdb->prepare( "SELECT %s as value from %s where post_id=%d;", $column, $table, $this->post->ID ) );
```
Comment (Gallmond) [inline includes/classes/sync/push.php:607]: one to watch

**Review by Gallmond — APPROVED** [review]

**Pull request merged**